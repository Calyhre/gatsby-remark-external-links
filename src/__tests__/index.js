const Remark = require(`remark`);
const plugin = require(`../index`);

const remark = new Remark().data(`settings`, {
  commonmark: true,
  footnotes: true,
  pedantic: true,
})

test(`with absolute link`, () => {
  const markdownAST = remark.parse(`[Test](https://example.com/path)`);
  plugin({ markdownAST });
  expect(markdownAST).toMatchSnapshot();
});

test(`with relative link`, () => {
  const markdownAST = remark.parse(`[Test](/path)`);
  plugin({ markdownAST });
  expect(markdownAST).toMatchSnapshot();
});
