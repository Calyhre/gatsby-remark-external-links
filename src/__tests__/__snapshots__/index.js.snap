// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with absolute link 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 6,
                  "line": 1,
                  "offset": 5,
                },
                "indent": Array [],
                "start": Object {
                  "column": 2,
                  "line": 1,
                  "offset": 1,
                },
              },
              "type": "text",
              "value": "Test",
            },
          ],
          "data": Object {
            "hProperties": Object {
              "rel": "nofollow noopener noreferrer",
              "target": "_blank",
            },
          },
          "position": Position {
            "end": Object {
              "column": 33,
              "line": 1,
              "offset": 32,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "title": null,
          "type": "link",
          "url": "https://example.com/path",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 33,
          "line": 1,
          "offset": 32,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 33,
      "line": 1,
      "offset": 32,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`with relative link 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 6,
                  "line": 1,
                  "offset": 5,
                },
                "indent": Array [],
                "start": Object {
                  "column": 2,
                  "line": 1,
                  "offset": 1,
                },
              },
              "type": "text",
              "value": "Test",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 14,
              "line": 1,
              "offset": 13,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "title": null,
          "type": "link",
          "url": "/path",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
